tests/func-tests/node_placement_test.go

```diff
@@ -0,0 +1,162 @@
+				//"kube-multus":     false,
+				"nmstate-handler": false,
+				"ovs-cni-marker":  false,
+				"virt-handler":    false,
+			}
+
+			var cnaoCR networkaddonsv1.NetworkAddonsConfig
+
+			s := scheme.Scheme
+			_ = apis.AddToScheme(s)
+			s.AddKnownTypes(networkaddonsv1.SchemeGroupVersion)
+			opts := k8smetav1.GetOptions{}
+			err = client.RestClient().Get().
+				Resource("networkaddonsconfigs").
+				Name("cluster").
+				VersionedParams(&opts, scheme.ParameterCodec).
+				Timeout(10 * time.Second).
+				Do().Into(&cnaoCR)
+
+			if cnaoCR.Spec.Ovs == nil {
+				delete(expectedWorkloadsPods, "ovs-cni-marker")
+			}
+
+			pods, err := client.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(k8smetav1.ListOptions{
+				FieldSelector: fmt.Sprintf("spec.nodeName=%s", workloadsNode.Name),
+			})
+
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, pod := range pods.Items {
+				podName := pod.Spec.Containers[0].Name
+				fmt.Fprintf(GinkgoWriter, "Found %s pod '%s' in the 'workloads' node %s\n", podName, pod.Name, workloadsNode.Name)
+				if found, ok := expectedWorkloadsPods[podName]; ok {
+					if !found {
+						expectedWorkloadsPods[podName] = true
+					}
+				}
+			}
+
+			Expect(expectedWorkloadsPods).ToNot(ContainElement(false))
+		})
+	})
+
+	Context("validate node placement on infra nodes", func() {
+		It("[test_id:5678] all expected 'infra' pod must be on infra node", func() {
+			infraNodes, err := client.CoreV1().Nodes().List(k8smetav1.ListOptions{
+				LabelSelector: "node.kubernetes.io/hco-test-node-type==infra",
+			})
+
+			Expect(err).ShouldNot(HaveOccurred())
+
+			expectedInfraPods := map[string]bool{
+				"cdi-apiserver":        false,
+				"cdi-controller":       false,
+				"cdi-uploadproxy":      false,
+				"manager":              false,
+				"nmstate-webhook":      false,
+				"virt-api":             false,
+				"virt-controller":      false,
+				"vm-import-controller": false,
+			}
+
+			for _, node := range infraNodes.Items {
+				pods, err := client.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(k8smetav1.ListOptions{
+					FieldSelector: fmt.Sprintf("spec.nodeName=%s", node.Name),
+				})
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, pod := range pods.Items {
+					podName := pod.Spec.Containers[0].Name
+					fmt.Fprintf(GinkgoWriter, "Found %s pod '%s' in the 'infra' node %s\n", podName, pod.Name, node.Name)
+					if found, ok := expectedInfraPods[podName]; ok {
+						if !found {
+							expectedInfraPods[podName] = true
+						}
+					}
+				}
+			}
+
+			Expect(expectedInfraPods).ToNot(ContainElement(false))
+		})
+	})
+
+	Context("validate node placement for vmi", func() {
+		It("[test_id:5679] should create, verify and delete VMIs with correct node placements", func() {
+			// we are iterating many times to ensure that the vmi is not scheduled to the expected node by chance
+			// this test may give false positive result. more iteration can give more accuracy.
+			testAmount := 20
```

Comment:
please use `const` instead of defining a variable.

Reply:
done